### DISCUSSION TITLE
Alter topic configurations

### DISCUSSION THREAD

**Discussion by m-kay** [spring-projects/spring-kafka#2258] [Q&A]
Is there a possibility to alter topic configurations in a similar way we can create new topics just by adding a `NewTopic` bean? 
As far as I could see the configurations in the `NewTopic` beans are only used to create topics and increase partition size but when setting a configuration like `retention.ms` this will only be used to initially create the topic.

**Comment by artembilan** [comment]


**Reply by garyrussell** [reply]
Actually, we do increase the number of partitions if we find the existing topic has fewer partitions. There is an [open issue](https://github.com/spring-projects/spring-kafka/issues/1482) about increasing the `retention.ms`. The `AdminClient` does support such an operation. As I said there, contributions are welcome.

**Reply by m-kay** [reply]
Thats exactly what I was searching for. There is already a pull request open for quite long, are you planing to merge that someday or what is missing there?

**Reply by garyrussell** [reply]
It just hasn't been given any priority, given that there is an easy work around with just a few lines of code. I have marked it as "ideal-for-user-contribution".

**Reply by garyrussell** [reply]
One concern I would have with doing it automatically every time an application starts is you might have 2 apps using the same topic, but with different configs and it would keep toggling back and forth.

**Reply by m-kay** [reply]
But isn't it already fully implemented in pull request #1823? Or should we may add a configuration which has to activate re-configuration explicitly?

**Reply by garyrussell** [reply]
Oh, sorry; I completely forgot about that PR; the contributors never responded to the review comments.

I'll take another look at it.

**Reply by m-kay** [reply]
Could you already have a look at it? Is there something I could do for this PR to be merged? I would really like to use this feature.
